### ISSUE TITLE
docker run issue

### ISSUE THREAD

**Issue by rocklinsuv** [danielsnider/MapViz-Tile-Map-Google-Maps-Satellite#3]
Hi danielsnider,

I've been trying to get MapViz running along with MapProxy according to your tutorial. When I tried to run "sudo docker run -p 8080:8080 -d -t -v ~/mapproxy:/mapproxy danielsnider/mapproxy", it initially downloaded the docker image, then the second time returned a chain of numbers and characters like "5d23e78.......". I then check the MapProxy demo http://127.0.0.1:8080/demo but nothing happened. So I would like to know if we need to install MapProxy individually before using the docker image? I would also like to know what is the code used for from this url: https://github.com/danielsnider/docker-mapproxy-googlemaps. Do I need to download it and use it along with docker?

Another issue will be I'm currently living in a country without Google (sucks). Is there an alternative WTMS with terrain maps rather than Google Maps? Any thoughts will be very much appreciated!

Best Regards,
rocklinsuv

**Comment by danielsnider** [comment]
Hi rocklinsuv,

Does the command "docker ps" show that the docker container is running? Also, docker logs <container id> will show you all the output of the conatiner run, can you share it here?

This container approach does not require a separate installation of mapproxy. 

To use a different map service than google maps edit the file mapproxy.yaml in your ~/mapproxy folder and relaunch the container. I think there is a terrain tilemap from Microsoft Bing but I don't have the URL right now and it might not even need format conversion depending on your use case. Search online and I can help you look if you don't find it.

**Comment by rocklinsuv** [comment]
Hi danielsnider,

I've captured a screenshot of the "docker run" command output:
![image](https://user-images.githubusercontent.com/7424627/30755992-3610ff80-9ffb-11e7-811d-f89cd759de2e.png)


I used "docker ps" but getting nothing. I also realized that I'm using a 32bit Ubuntu Mate 16.04, so the docker is docker.io rather than docker-ce. Do you think this is causing the issue? Since there is no container running, I could not use the "docker logs" command. Here's what I got from "docker info":

Containers: 6
 Running: 0
 Paused: 0
 Stopped: 6
Images: 2
Server Version: 1.12.6
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 27
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge overlay null
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.10.0-35-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: i686
CPUs: 2
Total Memory: 1.435 GiB
Name: ubuntu
ID: ZDK7:TDC6:LTXF:NT76:XXT3:H4O2:XJCA:LXN4:2ICI:DWG7:DOD6:TF2O
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8

And the "docker images" command shows:
REPOSITORY              TAG                 IMAGE ID            CREATED             SIZE
hello-world               latest              abd130ec0722        6 weeks ago         664.6 kB
danielsnider/mapproxy   latest       571a6aeb022d        7 months ago        1.031 GB

I'm also trying to google (I did manage to use Google but on my Windows machine, and the Ubuntu was running on a virtual machine) why the container won't start. So any help is very much appreciated ^_^

Update:
I downloaded the "ubuntu" images and successfully ran it. The "docker ps" command is giving me:

CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
16deded98e59        ubuntu              "/bin/bash"         13 seconds ago      Up 12 seconds                           sharp_engelbart


Cheers,
rocklinsuv

**Comment by danielsnider** [comment]
Do a "docker ps -a" this will show all containers even stopped ones, find
the id for the mapproxy container that exited, and use the id with "docker
logs <id>" and let me know :-).



Daniel Snider ツ

**Comment by rocklinsuv** [comment]
Okay, when I "docker ps -a", it showed:

![image](https://user-images.githubusercontent.com/7424627/30756624-0f1424cc-9ffe-11e7-95c9-a53da0970e02.png)


So I "docker logs 3e0a5a21d387", it gave me:

![image](https://user-images.githubusercontent.com/7424627/30756645-22dd15cc-9ffe-11e7-838b-f98c43650cef.png)

**Comment by danielsnider** [comment]
There's an error message in there but I'm not familiar with it. I would
suggest start small, does the hello world docker image work for you? Does a
simple python container work? docker run -it --rm python:2.7 bash

If these work and mapproxy doesn't then start with an empty Dockerfile and
add lines of code from the mapproxy container Dockerfile (from
https://github.com/danielsnider/docker-mapproxy-googlemaps/blob/master/Dockerfile)
until
you isolate where things start failing.

Daniel Snider ツ

**Comment by rocklinsuv** [comment]
Both hello-world and python:2.7 bash were running successfully. It is a bit late now (2am) so I'll try your method tomorrow. Thanks again for all your help!

**Comment by danielsnider** [comment]
Thank you too for the help.

Daniel Snider ツ

**Comment by rocklinsuv** [comment]
Hi Daniel,

It seems like there's a pip-install issue:
![image](https://user-images.githubusercontent.com/7424627/30791080-038dd0ae-a165-11e7-8cd1-2857c25cb3e1.png)

Everything was fine until the "pip install" command. I also googled the "cannot import name _remove_dead_weakref" error. It said it was some kinda python version issue, but still I have no clue how to solve this. 

Any thoughts? Thanks again for all your help :+1: 

Cheers,

**Comment by danielsnider** [comment]
Nice find. Can you isolate which package (out of Shapely Pillow MapProxy or
uwsgi) is causing the error? or is it any package?



Daniel Snider ツ

**Comment by rocklinsuv** [comment]
I believe it was any package. So it was the "pip install" which failed.

I even added a line "RUN pip install" before the original "RUN pip install Shapely" line and still got this error message:
![image](https://user-images.githubusercontent.com/7424627/30791682-f84ecc62-a168-11e7-8811-59b6d6804ced.png)

**Comment by danielsnider** [comment]
Oh pip =(.

I'm not sure but maybe you can update your base layer of docker's python,
the docker python 2.7 container, then maybe this bug will go away? I'm not
sure how to do that.

Maybe try all this on a new machine. A fresh install might not see this
problem.

Daniel Snider ツ

**Comment by rocklinsuv** [comment]
I'll see what I can do and keep you updated :D. Thanks!


________________________________

**Comment by rocklinsuv** [comment]
Hi Daniel,

I'm so glad that I've resolved the problem. I was using Ubuntu Mate 16.04 32bit with docker.io, so now I switched to Ubuntu Mate 16.04 64bit with docker-ce and the error is GONE! Now I'm re-installing ROS to test MapViz :D.


Best Regards,
rocklinsuv

**Comment by rocklinsuv** [comment]
Okay, it's done! Thank you so much for your help :+1: 
![screenshot at 2017-09-25 09 17 36](https://user-images.githubusercontent.com/7424627/30839067-0f8b3b42-a2a2-11e7-83eb-4a2898158ef1.png)

Also, I'm wondering if we could install MapProxy individually (not in docker) and achieve the same thing? I suppose we just need to modify the yaml file to set the ports correctly. Is it correct?

**Comment by danielsnider** [comment]
Yes you could achieve the same thing without docker, just follow mapproxy
setup steps found in this file
https://github.com/danielsnider/docker-mapproxy-googlemaps/blob/master/Dockerfile

Daniel Snider ツ

**Comment by rocklinsuv** [comment]
Hi Daniel,

It's me again. I've encountered a seeding issue so I would like to see you have any thought. My seed.yaml file looks like this:

seeds:
  myseed1:
    caches: [gm_cache]
    grids:  [gm_grid]
    coverages: [act]
    levels:
      to: 16
    refresh_before:
      time: 2013-10-10T12:35:00
coverages:
  act:
    datasource: 'shp/ACT_STATE_POLYGON_shp.shp'
    srs: 'EPSG:4283'

The shape file contains the boundary of Australian Captial Territory in EPSG 4283 coordinate. I managed to cache the entire ACT area using the mapproxy-seed -f mapproxy.yaml -s seed.yaml -c 6 command. After seeding, I launched MapViz, zoomed in to the ACT area, disconnected the internet to see if  MapViz will load the map from MapProxy local cache. However, the map looks like empty. I did check the cache_data folder and it contains the PNG files (several MBs of data). I'm wondering if there's anything I've done wrong, maybe the srs:EPSG stuff?. Any thought would be very much appreciated!
BTW, I'm not using docker but directly porting MapProxy to MapViz.

Cheers,
rocklinsuv

**Comment by rocklinsuv** [comment]
LMAO, I'm such an idiot. A simple click on the "Save" button solved all the problem lol. Sorry to bother you again :D
